allennlp/models/simple_tagger.py

```diff
@@ -82,24 +82,26 @@ def __init__(
             "encoder input dim",
         )
 
-        # We keep calculate_span_f1 as a constructor argument for API consistency with
-        # the CrfTagger, even it is redundant in this class
-        # (label_encoding serves the same purpose).
-        if calculate_span_f1 and not label_encoding:
-            raise ConfigurationError(
-                "calculate_span_f1 is True, but no label_encoding was specified."
-            )
         self.metrics = {
             "accuracy": CategoricalAccuracy(),
             "accuracy3": CategoricalAccuracy(top_k=3),
         }
 
-        if calculate_span_f1 or label_encoding:
+        # We keep calculate_span_f1 as a constructor argument for API consistency with
+        # the CrfTagger, even it is redundant in this class
+        # (label_encoding serves the same purpose).
+        if calculate_span_f1 is None:
+            calculate_span_f1 = label_encoding is not None
+
+        self.calculate_span_f1 = calculate_span_f1
+        if calculate_span_f1:
+            if not label_encoding:
+                raise ConfigurationError(
+                    "calculate_span_f1 is True, but no label_encoding was specified."
+                )
             self._f1_metric = SpanBasedF1Measure(
                 vocab, tag_namespace=label_namespace, label_encoding=label_encoding
             )
-        else:
```

Comment:
I think we should keep this "else" clause where `self._f1_metric` is set to `None` so that it's always defined.

Reply:
Oh sorry, I removed `else: self._f1_...` based on `CrfTagger`.

https://github.com/allenai/allennlp-models/blob/master/allennlp_models/tagging/models/crf_tagger.py#L138-L145

I reverted the change in 51012f8.